### PR TITLE
Add build files for bazel

### PR DESCRIPTION
**Pull request by cgilmour**
This PR adds a bazel workspace and build files.
Other projects that use bazel for their build / dependency tool can then add this library to their project.

To build using bazel:
```
git clone ...
cd dd-opentracing-cpp
bazel build //:dd_opentracing_cpp
```

TODO:
- add build instructions to a readme.
- add instructions for using as a bazel dependency.
- add to circleci config.

Those will be done after the current task that intends to use this library in a different project.

### CODE REVIEW

File: BUILD.bazel [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    visibility = ["//visibility:public"],
+    deps = [
+        "//:3rd_party_nlohmann",
+        "@io_opentracing_cpp//:opentracing",
+        "@com_github_msgpack_msgpack_c//:msgpack",
+    ],
+    copts = ["-std=c++14"],
+)
+
+genrule(
```
Comment (willgittoes-dd) [inline BUILD.bazel]: ahaha this was such cmake-specific bastardry that you had to replicate 😅

File: bazel/external/msgpack.BUILD [hunk 2]
```diff
@@ -0,0 +1,15 @@
+cc_library(
+    name = "msgpack",
+    srcs = glob([
+        "src/*.c",
+        "include/**/*.h",
+        "include/**/*.hpp",
+    ]),
+    # hdrs = glob([
```
Comment (willgittoes-dd) [inline bazel/external/msgpack.BUILD]: Remove commented

File: BUILD.bazel (hunk 1 above)
Comment (cgilmour) [reply BUILD.bazel]: Yup, inspired by lightstep's bazel setup.

File: bazel/external/msgpack.BUILD (hunk 2 above)
Comment (cgilmour) [reply bazel/external/msgpack.BUILD]: Removed in followup commit.

**Comment by cgilmour** [comment]
The TODOs from the opening comment have been put in a backlog card so they can be picked up later on.

**Pull request merged**